File: ckgame-java/src/main/java/com/ckgame/military/WarParticipant.java
package com.ckgame.military;

import com.ckgame.core.calendar.GameDate;

/**
 * 战争参与者。
 */
public final class WarParticipant {
    public int character;
    public boolean isAttacker;
    public GameDate joined;
    public double contribution = 0.0;

    public WarParticipant(int character, boolean isAttacker, GameDate joined) {
        this.character = character;
        this.isAttacker = isAttacker;
        this.joined = joined;
    }
}

File: ckgame-ts/src/military/WarParticipant.ts
import { GameDate } from '../core/calendar/GameDate.js';

/**
 * A participant in a war.
 */
export class WarParticipant {
  public character: number;
  public isAttacker: boolean;
  public joined: GameDate;
  public contribution: number = 0;

  constructor(character: number, isAttacker: boolean, joined: GameDate) {
    this.character = character;
    this.isAttacker = isAttacker;
    this.joined = joined;
  }
}
